app/src/main/java/com/distillery/interview/ui/current_weather/CurrentWeatherFragment.kt

```diff
@@ -0,0 +1,73 @@
+package com.distillery.interview.ui.current_weather
+
+import android.os.Bundle
+import android.view.View
+import android.widget.Toast
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import com.distillery.interview.R
+import com.distillery.interview.data.models.Result
+import com.distillery.interview.data.models.WeatherResponse
+import com.distillery.interview.data.source.WeatherRepository
+import com.distillery.interview.data.source.remote.WeatherRemoteDataSource
+import com.distillery.interview.databinding.FragmentCurrentWeatherBinding
+
+class CurrentWeatherFragment : Fragment(R.layout.fragment_current_weather) {
+
+    private val viewModel: CurrentWeatherViewModel by viewModels {
+        CurrentWeatherViewModel.Factory(
+            WeatherRepository(
+                WeatherRemoteDataSource
```

Comment:
Is this working? I'm not seeing any constructor invocation here for `WeatherRemoteDataSource`.

Reply:
It works in this case because `WeatherRemoteDataSource` is an object type. So, it have no constructor, I only use the property access to pass the instance.

Reply:
Change it to be a class, please and thanks.